src/pages/signin.tsx

```diff
@@ -3,9 +3,21 @@ import Head from 'next/head';
 import Link from 'next/link';
 import { type NextPage } from 'next';
 import { SignInForm } from '../components/SignIn/Form';
-// import { SignInSocial } from '../components/SignIn/Social';
+import { useEffect } from 'react';
+import Router from 'next/router';
+import { type UserProps } from '../lib/types';
+
+const SignIn: NextPage<UserProps> = (props: UserProps) => {
```

Comment:
I don't think we need to specify `UserProps` twice here.

Reply:
And we could also destructure the user:
```js
const SignIn: NextPage<UserProps> = ({ user }: UserProps) => {
```
That way won't need the below line, i.e.
```js
const user = props.user;
```

Reply:
done